src/Analysis/Ast/Impl/Types/PythonClassType.cs

```diff
@@ -57,6 +53,17 @@ internal PythonClassType(string name, Location location)
             location.Module.AddAstNode(this, classDefinition);
         }
 
+        public override string Name {
+            get {
+                var genericParams = GenericParameters.Values.ToList();
```

Comment:
This should be `{get; private set;}` assigned right when parameters are filled, no need for delayed construction - which also may end up having threading issues.

Reply:
Ok I agree.